service/ui_plugin_build_baron.go

```diff
@@ -139,102 +181,135 @@ func (uis *UIServer) bbGetCreatedTickets(w http.ResponseWriter, r *http.Request)
 	gimlet.WriteJSON(w, results)
 }
 
-func (uis *UIServer) bbJiraSearch(rw http.ResponseWriter, r *http.Request) {
+// Retrieve the user feedback for a task id that has been stored by
+// the BF Suggestion Server.
+func (uis *UIServer) bbGetFeedback(rw http.ResponseWriter, r *http.Request) {
 	vars := gimlet.GetVars(r)
 	taskId := vars["task_id"]
 	exec := vars["execution"]
-	oldId := fmt.Sprintf("%v_%v", taskId, exec)
-	t, err := task.FindOneOld(task.ById(oldId))
+
+	// grab the user info
+	u := MustHaveUser(r)
+
+	t, bfsc, err := uis.bbGetTaskAndBFSuggestionClient(taskId, exec)
 	if err != nil {
 		gimlet.WriteJSONInternalError(rw, err.Error())
 		return
 	}
-	// if the archived task was not found, we must be looking for the most recent exec
-	if t == nil {
-		t, err = task.FindOne(task.ById(taskId))
-		if err != nil {
-			gimlet.WriteJSONInternalError(rw, err.Error())
-			return
-		}
-	}
 
-	bbProj, ok := uis.buildBaronProjects[t.Project]
-	if !ok {
-		gimlet.WriteJSON(rw, fmt.Sprintf("Corresponding JIRA project for %v not found", t.Project))
+	feedbackItems, err := bfsc.getFeedback(context.TODO(), t, u.Username())
```

Comment:
request objects have contexts so you can get one with r.Context()

Reply:
Good to know. Done.